### ISSUE TITLE
ERROR: Code not generated for probe: hardware:cache-misses:1000000

### ISSUE THREAD

**Issue by seudut** [iovisor/bpftrace#2776]
### What reproduces the bug?

```
root@my:/home/my/project/scripts# ./bin/bpftrace -e 'hardware:cache-misses:1000000,tracepoint:raw_syscalls:sys_enter {@[probe]=count()} interval:s:5 {print(@); clear(@);}'
Attaching 3 probes...
ERROR: Code not generated for probe: hardware:cache-misses:1000000 from: hardware:cache-misses:1000000,tracepoint:raw_syscalls:sys_enter
```
it seems `cache-misses` not support the built variable `probe` ?

### bpftrace --info

bpftrace v0.18.0

**Comment by jordalgo** [comment]
@seudut That script works for me. What hardware and kernel version are you running this on?

**Comment by seudut** [comment]
> @seudut That script works for me. What hardware and kernel version are you running this on?

OS is  debian, kernel 5.10.178-3
cpu is intel i7

here is the information

```
root@peng:/home/peng/project/scripts/bin# ./bpftrace -e 'hardware:cache-misses:1000000,tracepoint:raw_syscalls:sys_enter {@[probe]=count()} interval:s:5 {print(@); clear(@);}'
Attaching 3 probes...
ERROR: Code not generated for probe: hardware:cache-misses:1000000 from: hardware:cache-misses:1000000,tracepoint:raw_syscalls:sys_enter
root@peng:/home/peng/project/scripts/bin# uname -a
Linux peng 5.10.0-22-amd64 #1 SMP Debian 5.10.178-3 (2023-04-22) x86_64 GNU/Linux
root@peng:/home/peng/project/scripts/bin# lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          12
On-line CPU(s) list:             0-11
Thread(s) per core:              2
Core(s) per socket:              6
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           158
Model name:                      Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
Stepping:                        10
CPU MHz:                         4331.413
CPU max MHz:                     4700.0000
CPU min MHz:                     800.0000
BogoMIPS:                        7399.70
Virtualization:                  VT-x
L1d cache:                       192 KiB
L1i cache:                       192 KiB
L2 cache:                        1.5 MiB
L3 cache:                        12 MiB
NUMA node0 CPU(s):               0-11
Vulnerability Itlb multihit:     KVM: Mitigation: VMX disabled
Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:          Mitigation; IBRS
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; IBRS, IBPB conditional, STIBP conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Mitigation; Microcode
Vulnerability Tsx async abort:   Mitigation; Clear CPU buffers; SMT vulnerable
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx
                                 est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 sm
                                 ep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d
```





looks like not only `cache-misses`, but also `software:context-switches`. while others tracepoint like `syscalls` is ok

```
root@peng:/home/peng/project/scripts/bin# ./bpftrace -e 'hardware:cache-misses:1000000 {@[probe]=count()} interval:s:5 {print(@); clear(@);}'
Attaching 2 probes...
ERROR: Code not generated for probe: hardware:cache-misses:1000000
root@peng:/home/peng/project/scripts/bin# ./bpftrace -e 'software:context-switches:1000 {@[probe]=count()} interval:s:5 {print(@); clear(@);}'
Attaching 2 probes...
ERROR: Code not generated for probe: software:context-switches:1000
root@peng:/home/peng/project/scripts/bin# ./bpftrace -e 'tracepoint:syscalls:sys_enter* {@[probe] = count()} interval:s:5 {print(@, 5); clear(@);}'
Attaching 337 probes...
@[tracepoint:syscalls:sys_enter_read]: 1283
@[tracepoint:syscalls:sys_enter_poll]: 2401
@[tracepoint:syscalls:sys_enter_epoll_wait]: 3252
@[tracepoint:syscalls:sys_enter_recvmsg]: 6160
@[tracepoint:syscalls:sys_enter_futex]: 29140
^C^C@[tracepoint:syscalls:sys_enter_read]: 2568
@[tracepoint:syscalls:sys_enter_write]: 3357
@[tracepoint:syscalls:sys_enter_poll]: 6090
@[tracepoint:syscalls:sys_enter_recvmsg]: 10881
@[tracepoint:syscalls:sys_enter_futex]: 17076
```

**Comment by jordalgo** [comment]
Sorry, can you also run `perf list` to see if they show up there?

**Comment by seudut** [comment]
> Sorry, can you also run `perf list` to see if they show up there?

yes,

```
List of pre-defined events (to be used in -e):

  branch-instructions OR branches                    [Hardware event]
  branch-misses                                      [Hardware event]
  bus-cycles                                         [Hardware event]
  cache-misses                                       [Hardware event]
  cache-references                                   [Hardware event]
  cpu-cycles OR cycles                               [Hardware event]
  instructions                                       [Hardware event]
  ref-cycles                                         [Hardware event]

  alignment-faults                                   [Software event]
  bpf-output                                         [Software event]
  context-switches OR cs                             [Software event]
  cpu-clock                                          [Software event]
  cpu-migrations OR migrations                       [Software event]
  dummy                                              [Software event]
  emulation-faults                                   [Software event]
  major-faults                                       [Software event]
  minor-faults                                       [Software event]
  page-faults OR faults                              [Software event]
  task-clock                                         [Software event]

  duration_time                                      [Tool event]
```

**Comment by viktormalik** [comment]
This has been fixed by 817179075c67d9d55a10661ec44047eab1e4e52a. Upgrading to v0.19 should resolve the issue.
